app/src/main/kotlin/io/homeassistant/companion/android/websocket/WebsocketBroadcastReceiver.kt

```diff
@@ -3,9 +3,17 @@ package io.homeassistant.companion.android.websocket
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
 
 class WebsocketBroadcastReceiver : BroadcastReceiver() {
+    private val ioScope: CoroutineScope = CoroutineScope(Dispatchers.IO + Job())
+
     override fun onReceive(context: Context, intent: Intent) {
-        WebsocketManager.start(context)
+        ioScope.launch {
+            WebsocketManager.start(context)
+        }
```

Comment:
@jpelgrom I think we should do something about all these scopes that we are created in the app without a lifecycle attached. We should maybe use the GlobalScope instead of creating all these scopes (since it is basically the same), or at least a wrapper around the GlobalScope and avoid using it whenever we can...

I also think we could make something like this
```kotlin
abstract class CoroutineBroadcastReceiver : BroadcastReceiver() {
    final override fun onReceive(context: Context, intent: Intent) {
        GlobalScope.launch {
            withTimeout(10.seconds) {
                onReceiveIntent(context, intent)
            }
        }
    }

    abstract suspend fun onReceiveIntent(context: Context, intent: Intent)
}
```
In the same spirit as CoroutineWorker

Reply:
This one is more an audit I would say https://github.com/home-assistant/android/issues/6001